src/Catel.Core/Catel.Core.Shared/Data/ModelBase.propertyediting.cs

```diff
@@ -122,7 +122,7 @@ internal void SetValue(PropertyData property, object value, bool notifyOnChange,
                 }
 
                 // Validate before assigning, dynamic properties will cause exception
-                if (validateAttributes && !LeanAndMeanModel)
+                if (validateAttributes && !LeanAndMeanModel && ValidateUsingDataAnnotations)
```

Comment:
Why not do this check inside ValidatePropertyUsingAnnotations? We also have a "SuspendValidation" check there.